File: ProjectOne/proj1_a.cpp
//Morgan Monzingo
//Part a

#include <iostream>
#include "mat.h"
#include "nest.cpp"
#include <cmath>
using namespace std;

int main()
{
	//creating matrices and initializing them to size of z
	Mat z = Linspace(-4, 4, 801);
	double value = 0.0;
	Mat p5(801);
	Mat p9(801);
	Mat p13(801);
	Mat f(801);
	Mat err5(801);
	Mat err9(801);
	Mat err13(801);

	//creating arrays to then put into the corresponding matrix
	double tempp_5[] = {0, 1.0, 0, -1.0/6.0, 0, 1.0/120.0};
	Mat p_5 = Mat(6 , 1 , tempp_5);
	double tempp_9[] = {0, 1.0, 0, -1.0/6.0, 0, 1.0/120.0, 0, -1.0/5040.0, 0, 1.0/362880.0};
	Mat p_9 = Mat(10, 1, tempp_9);
	double tempp_13[] = {0, 1.0, 0, -1.0/6.0, 0, 1.0/120.0, 0, -1.0/5040.0, 0, 1.0/362880.0, 0, -1.0/39916800.0, 0, 1.0/6227020800.0};
	Mat p_13 = Mat(14, 1, tempp_13);

	//use the z matrix to solve polynomial using nest function and z as the x value
	 for(int counter = 0; counter < z.Size(); counter++)
	 {
		p5(counter) = nest(p_5, z(counter)); 
		p9(counter) = nest(p_9, z(counter)); 
		p13(counter) = nest(p_13,z(counter));

		f(counter) = sin(z(counter));

		err5(counter) = fabs(f(counter)-p5(counter));
		err9(counter) = fabs(f(counter)-p9(counter));
		err13(counter) = fabs(f(counter)-p13(counter));
	 }

	 //writing all matrices into files 
	z.Write("z.txt");
	p5.Write("p5.txt");
	p9.Write("p9.txt");
	p13.Write("p13.txt");
	f.Write("f.txt");
	err5.Write("err5.txt");
	err9.Write("err9.txt");
	err13.Write("err13.txt");





	
	return 0;
}

File: ProjectOne/proj1_a.py
#!/usr/bin/env python
#Morgan Monzingo
#Project 1 
#9.8.14

import numpy as np
import matplotlib.pyplot as plt

#load all files into variables

z = np.loadtxt("z.txt")
f = np.loadtxt("f.txt")
p5 = np.loadtxt("p5.txt")
p9 = np.loadtxt("p9.txt")
p13 = np.loadtxt("p13.txt")
err5 = np.loadtxt("err5.txt")
err9 = np.loadtxt("err9.txt")
err13 = np.loadtxt("err13.txt")

#set graph one as the standard plad of f,p5,p9,p13 vs z
graph1 = plt.figure(1)
graph1 = plt.plot(z,f, 'b--', label="sin(x)")
graph1 = plt.plot(z,p5, 'r', label="p5")
graph1 = plt.plot(z,p9, 'm', label="p9")
graph1 = plt.plot(z,p13,'k',  label="p13")
graph1 = plt.legend()
graph1 = plt.ylabel("value of polynomial")
graph1 = plt.xlabel("z")
graph1 = plt.title("sin(x) taylor series")
plt.show()

#plot the semi#logy graph of all of the error
graph2 = plt.figure(2)
graph2 = plt.semilogy(z,err5, label = "err5")
graph2 = plt.semilogy(z,err9, label = "err9")
graph2 = plt.semilogy(z,err13, label = "err13")
graph2 = plt.legend()
graph2 = plt.ylabel("error")
graph2 = plt.xlabel("z")
graph2 = plt.title("Error")
plt.show()


